src/app/posts/new/page.tsx

```diff
@@ -2,10 +2,16 @@
 import CreatePostForm from '@/components/posts/CreatePostForm';
 
 interface NewPostPageProps {
-  searchParams: { boardId: number };
+  searchParams: Promise<{ boardId?: number }>;
```

Comment:
`?`가 있으면 타입만 봤을 때는 `boardId`가 생략 가능한 걸로 보이게 돼요.
```suggestion
  searchParams: Promise<{ boardId: number }>;
```